### PR TITLE
Roles admin REST API: Don't expand composite roles

### LINKED ISSUE #26951: Roles admin REST API for creating roles: Composite roles are expanded
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/api

### Describe the bug

When using the `roles` admin REST API for creating a composite role that contains other composite roles then the contained roles are expanded. This is different from the behaviour when adding composite roles in the Admin UI and the behaviour of `POST {role-name}/composites`. This is unexpected, makes managing roles harder and is probably not intended.

### Version

23.0.6

### Expected behavior

Composite roles added to a composite role are not expanded.

### Actual behavior

Composite roles added to a composite role are expanded.

### How to Reproduce?

Create some roles with composites: `role3 contains (role2 contains role1)`

```
POST /auth/admin/realms/master/roles
 
{
    "name": "role1"
}
 
POST /auth/admin/realms/master/roles
 
{
    "composite": true,
    "name": "role2",
    "composites": {
        "realm": ["role1"]
    }
}
 
POST /auth/admin/realms/master/roles
 
{
    "composite": true,
    "name": "role3",
    "composites": {
        "realm": ["role2"]
    }
}
```

`role1` was added because `role2` was expanded. (It is not expected to have `role1` added explicitly to `role3` as it is already a inherited role from `role2`.)

``` 
GET /auth/admin/realms/master/roles/role3/composites
 
[
    {
        "id": "981a8351-b895-47ad-8923-b94639a47148",
        "name": "role1",
        "composite": false,
        "clientRole": false,
        "containerId": "5f3e9387-7494-44ba-8699-758122ee07b3"
    },
    {
        "id": "fdd4f720-7016-49c8-aa8d-a9371e7472b6",
        "name": "role2",
        "composite": true,
        "clientRole": false,
        "containerId": "5f3e9387-7494-44ba-8699-758122ee07b3"
    }
]
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by synth3**
Closes #26951

Additionally `requireMapComposite` as in `RoleResource.addComposites` was added.

### CODE REVIEW

File: services/src/main/java/org/keycloak/services/resources/admin/RoleContainerResource.java [hunk 1]
```diff
@@ -179,7 +174,10 @@ public Response createRole(final RoleRepresentation rep) {
                         }
                         realmRoles.add(realmRole);
                     }
-                    RoleUtils.expandCompositeRoles(realmRoles).forEach(role::addCompositeRole);
+                    realmRoles.forEach(r -> {
```
Comment (pedroigor) [inline services/src/main/java/org/keycloak/services/resources/admin/RoleContainerResource.java:177]:
```suggestion
                   realmRoles.stream().peek(roles::requireMapComposite).forEach(role::addCompositeRole);
```

File: services/src/main/java/org/keycloak/services/resources/admin/RoleContainerResource.java [hunk 2]
```diff
@@ -200,7 +198,10 @@ public Response createRole(final RoleRepresentation rep) {
                             }
                             clientRoles.add(clientRole);
                         }
-                        RoleUtils.expandCompositeRoles(clientRoles).forEach(role::addCompositeRole);
+                        clientRoles.forEach(r -> {
```
Comment (pedroigor) [inline services/src/main/java/org/keycloak/services/resources/admin/RoleContainerResource.java:201]:
```suggestion
                        clientRoles.stream().peek(roles::requireMapComposite).forEach(role::addCompositeRole);
```

**Review by pedroigor — COMMENTED** [review]
@synth3 Thanks. Only a minor comment.

**Comment by pedroigor** [comment]
@mposolda Does it make sense to you?

**Review by mposolda — APPROVED** [review]
@pedroigor LGTM. Feel free to merge if you are ok with this change as well

**Pull request merged**